cloud/openstack/os_zone.py

```diff
@@ -0,0 +1,231 @@
+RETURN = '''
+zone:
+    description: Dictionary describing the zone.
+    returned: On success when I(state) is 'present'.
+    type: dictionary
+    contains:
+        id:
+            description: Unique zone ID
+            type: string
+            sample: "c1c530a3-3619-46f3-b0f6-236927b2618c"
+        name:
+            description: Zone name
+            type: string
+            sample: "example.net."
+        type:
+            description: Zone type
+            type: string
+            sample: "PRIMARY"
+        email:
+            description: Zone owner email
+            type: string
+            sample: "test@example.net"
+        description:
+            description: Zone description
+            type: string
+            sample: "Test description"
+        ttl:
+            description: Zone TTL value
+            type: int
+            sample: 3600
+        masters:
+            description: Zone master nameservers
+            type: list
+            sample: []
+'''
+
+
+def _system_state_change(state, email, description, ttl, masters, zone):
+    if state == 'present':
+        if not zone:
+            return True
+        if email is not None and zone.email != email:
+            return True
+        if description is not None and zone.description != description:
+            return True
+        if ttl is not None and zone.ttl != ttl:
+            return True
+        if masters is not None and zone.masters != masters:
+            return True
+    if state == 'absent' and zone:
+        return True
+    return False
+
+def main():
+    argument_spec = openstack_full_argument_spec(
+        name=dict(required=True),
+        zone_type=dict(required=False, choice=['primary', 'secondary']),
+        email=dict(required=False, default=None),
+        description=dict(required=False, default=None),
+        ttl=dict(required=False, default=None, type='int'),
+        masters=dict(required=False, default=None, type='list'),
+        state=dict(default='present', choices=['absent', 'present']),
+    )
+
+    module_kwargs = openstack_module_kwargs()
+    module = AnsibleModule(argument_spec,
+                           supports_check_mode=True,
+                           **module_kwargs)
+
+    if not HAS_SHADE:
+        module.fail_json(msg='shade is required for this module')
+    if StrictVersion(shade.__version__) < StrictVersion('1.8.0'):
+        module.fail_json(msg="To utilize this module, the installed version of"
+                             "the shade library MUST be >=1.8.0")
+
+    name = module.params.pop('name')
+    state = module.params.pop('state')
+
+    try:
+        cloud = shade.openstack_cloud(**module.params)
+        zone = cloud.get_zone(name)
+
+        if module.check_mode:
+            module.exit_json(changed=_system_state_change(state, email,
+                                                          description, ttl,
+                                                          masters, zone))
+
```

Comment:
Some of these parameters are not defined until further below, so this check will fail.